src/facade/facade_types.h

```diff
@@ -56,6 +60,23 @@ struct ConnectionStats {
   ConnectionStats& operator+=(const ConnectionStats& o);
 };
 
+struct ErrorReply {
+  explicit ErrorReply(std::string msg, std::string_view kind = {})
+      : message{move(msg)}, kind{kind} {
+  }
+  explicit ErrorReply(std::string_view msg, std::string_view kind = {}) : message{msg}, kind{kind} {
+  }
+  explicit ErrorReply(const char* msg, std::string_view kind = {})
+      : message{std::string_view{msg}}, kind{kind} {
+  }
+  explicit ErrorReply(OpStatus status) : message{}, kind{}, status{status} {
+  }
+
+  std::variant<std::string, std::string_view> message;
+  std::string_view kind;
+  std::optional<OpStatus> status{std::nullopt};
+};
+
```

Comment:
I don't feel strong about having such a heavy type inside facade_types (which is included almost everywhere, but not sure where else to put it)

Reply:
I know its not used everywhere where it fits in, I'll have more iterations when I check other parts

Reply:
The const char msg* overload is to avoid ambiguity (const char* converts to both string and string_view implicitly)

Reply:
Maybe in order to reduce ambiguity, the `std::string` c'tor could accept a `std::string&&` instead? (I imagine this will allow removing the `const char*` one, which is confusing).
If for some reason that doesn't work, I'd write that in a comment.

Reply:
🤷🏻 
![image](https://github.com/dragonflydb/dragonfly/assets/20553775/b70c4049-1a7a-482a-9207-1532e65422bb)